databuilder/publisher/neo4j_csv_publisher.py

```diff
@@ -318,10 +323,22 @@ def _publish_relation(self, relation_file: str, tx: Transaction) -> Transaction:
 
         with open(relation_file, 'r', encoding='utf8') as relation_csv:
             for rel_record in pandas.read_csv(relation_csv, na_filter=False).to_dict(orient="records"):
-                stmt = self.create_relationship_merge_statement(rel_record=rel_record)
-                params = self._create_props_param(rel_record)
-                tx = self._execute_statement(stmt, tx, params,
-                                             expect_result=self._confirm_rel_created)
+                badge_exception = True
```

Comment:
any we reason we need to make it specific for badge exception?  we just need to retry if deadlock/TransientError occurs?

Reply:
oh, it seems you update the exception handling loop. could you rename the variable a bit more generic?